Scripts/Sources/Automation/Eval.swift

```diff
@@ -0,0 +1,329 @@
+            try Shell.executeAndPrint("git -C \(dir) push \(remote) \(branch)", timeout: 30)
+        } else {
+            throw CIError.logicalError(message: "Repository URL not found")
+        }
+    }
+
+    static func runCocoaPodsLinter() throws {
+        print("🔮 Validating CocoaPods support")
+        let flags = TravisCI.isRunningLocally() ? "--verbose" : ""
+        try Shell.executeAndPrint("bundle exec pod lib lint \(flags)", timeout: 300)
+    }
+
+    static func testCoverage() throws {
+        defer {
+            print("📦 ✨ Cleaning up")
+            try! Shell.executeAndPrint("rm -f Eval.framework.coverage.txt")
+            try! Shell.executeAndPrint("rm -f EvalTests.xctest.coverage.txt")
+        }
+
+        print("☝🏻 Uploading code test coverage data")
+        try Shell.executeAndPrint("bash <(curl -s https://codecov.io/bash) -J Eval", timeout: 120)
+    }
+
+    static func runDanger() throws {
+        if TravisCI.isRunningLocally() {
+            print("⚠️ Running Danger in local mode")
+            try Shell.executeAndPrint("bundle exec danger local || true", timeout: 120)
+        } else if TravisCI.isPullRquestJob() {
+            print("⚠️ Running Danger")
+            try Shell.executeAndPrint("bundle exec danger || true", timeout: 120)
+        }
+    }
+
+    static func prepareExamplesForBuild() throws {
+        print("🤖 Generating project files for Examples")
+        try onAllExamples { _ in
+            let cleanup = [
+                "rm -f Package.resolved",
+                "rm -rf .build",
+                "rm -rf build"
+            ]
+            let build = [
+                "swift package generate-xcodeproj"
+            ]
+            return (cleanup + build).joined(separator: " && ")
+        }
+
+        try performManualSteps()
+    }
+
+    static func buildExamples() throws {
+        print("♻️ Building Examples")
+        try onAllExamples { example in
+            "xcodebuild clean build -scheme \(example)-Package | bundle exec xcpretty --color"
+        }
+    }
+
+    static func runTestsOnExamples() throws {
+        print("👀 Running automated tests on Examples")
+        try onAllExamples { example in
+            "xcodebuild test -scheme \(example)-Package | bundle exec xcpretty --color"
+        }
+    }
+
+    // MARK: Helpers
+
+    static func onAllExamples(_ command: (String) throws -> String) throws {
+        for (name, directory) in try examples() {
+            let commands = [
+                "pushd \(directory)",
+                try command(name),
+                "popd"
+            ]
+            try Shell.executeAndPrint(commands.joined(separator: " && "), timeout: 120)
+        }
+    }
+
+    static func examples() throws -> [(name: String, directory: String)] {
+        let directory = "Examples"
+        return try FileManager.default.contentsOfDirectory(atPath: directory).map { ($0, "\(directory)/\($0)") }.filter { !$0.name.hasPrefix(".") }
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/tevelee/Eval/blob/fc973589729700ac4c55183b359a3f3047c6157f/Scripts/Sources/Automation/Eval.swift#L267"></span>Line should be 120 characters or less: currently 147 characters</td>
    </tr>
  </tbody>
</table>